### DISCUSSION TITLE
内挂繁体字幕转简体

### DISCUSSION THREAD

**Discussion by zhengzavan** [hooke007/mpv_PlayKit#215] [Q&A 问答]
最近看动漫主要是baha的源，但是繁体字幕看着实在是不适应，但是在github找了好久都没找到内嵌繁体转简体的脚本，还是说mpv的脚本不足以支持该功能？

**Comment by hooke007** [comment]
内嵌字幕是硬字幕，是视频帧（画面）的一部分，要提取只能ocr
你是不是想说内封/内挂字幕？

这需求太小众了，mpv在windows也不是一个流行的播放器。在小圈子里的小圈子的需求，除了自己动手尝试就看运气有没有同道中人实现了。

**Reply by zhengzavan** [reply]
对就是内封字幕，我在GitHub 上面找到一个使用Crow 翻译的脚本，但是跑不起来。
https://github.com/EnergoStalin/subutils-mpv

**Reply by zhengzavan** [reply]
没尝试过关于mpv脚本的开发，但是感觉也就是 源字幕 字典 翻译后字幕，就是不知道mpv在获取和输出字幕上是如何实现的

**Reply by hooke007** [reply]
> You can extend it with you favorite translator by contributing one.

简单看了一下只有<s>linux支持和</s>英译俄。显然不是开箱即用的，需要贡献。

而且也不是“转换”，只是翻译原字幕然后创建一个新的osd渲染层输出字幕

**Reply by zhengzavan** [reply]
按作者的描述Windows 支持应该是没问题的，安装crow即可。如果Crow 使用了bing Google 等翻译引擎，支持其他语言应该也可以

**Reply by hooke007** [reply]
我已经修正描述了，但这个脚本不是你想要的。它更像是galgame的即时翻译工具一样的东西，额外创建一个OSD层显示文本

**Reply by zhengzavan** [reply]
确实，不过要是这个脚本能跑起来也能达到曲线救国的效果。

**Reply by zhengzavan** [reply]
用了一个偏门的方法，https://github.com/nowar-fonts/Nowar-Sans-Pseudo-Simplified  下载伪简体 bKAI00M.ttf 到fonts文件夹里，然后mpv.conf配置 sub-font = "~~/fonts/bKAI00M.ttf"。

**Reply by hooke007** [reply]
`sub-font` 可以在运行时更改，不影响你的使用习惯就行，否则我建议做input.conf方案避免初始字幕样式的修改。
```
<key> cycle-values sub-font "sans" "有爱黑体 GB18030 伪简体 Extended" ; script-binding input_plus/trackS_refresh
```
这玩意的fontname整这么长我真的活久见，后一个指令来自我仓库的脚本 input_plus.lua
